lua/def.c

```diff
@@ -597,7 +597,11 @@ const upb_FileDef* lupb_FileDef_check(lua_State* L, int narg) {
 
 static int lupb_FileDef_Dependency(lua_State* L) {
   const upb_FileDef* f = lupb_FileDef_check(L, 1);
+#if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM >= 501
+  int index = luaL_checkinteger(L, 2);
```

Comment:
Maybe these lines should all be changed to be `lupb_checkint32` instead? checkinteger() is a ptrdiff_t, so on 64-bit targets that means it'll be an int64 and 'int' is an int32 so this line is an implicit cast as you wrote it.

If _checkint32 is not appropriate for some reason, can you make a lupb_checkinteger() in upb.h/.c which just internally does the #ifdefined and uses check_int or check_integer based on which lua version supports it? It'll be cleaner than repeating the #ifs each time in this file (and have a place to have a comment about the large-ints overflow behavior)